pkg/kubelet/config/http.go

```diff
@@ -206,5 +206,9 @@ func applyDefaults(pod *api.Pod, url string) error {
 	// Always overrides the namespace.
 	pod.Namespace = kubelet.NamespaceDefault
 	glog.V(5).Infof("Using namespace %q for pod %q from URL %s", pod.Namespace, pod.Name, url)
+
+	// TODO(dchen1107): Remove when Kubelet uses pods directly.
+	pod.ObjectMeta.SelfLink = fmt.Sprintf("/api/v1beta2/pods/%s?namespace=%s",
```

Comment:
I remembered that I introduced this a while back, why we removed it at the first place?